python/pyspark/sql/functions.py

```diff
@@ -48,8 +55,11 @@ def _(col):
     return _
 
 
-def _create_function(name, doc=""):
-    """ Create a function that takes a Column object, by name"""
+def _create_function_with_column(name, doc=""):
```

Comment:
While I'm not opposed to renaming these to make their behaviour more clear, I find `_create_function_with_column` quite confusing.

Reply:
Which name would you like to suggest?

Reply:
`_create_column_function` or `_create_function_over_column`. Mathematically, if a function is defined over real numbers, we'd call it a "real function" or a "function over real numbers". This is the same principle.

Reply:
I think the classification used in this file is different. In this file, I see, for instance, binary functions, string functions, window functions which is closely related with SparkSQL's type, argument specification, or how it works, which is defined in Scala side's SQL side.

I already described the details in the doc test. I don't see big defferences between `_create_function_with_column`, `_create_column_function` and `_create_function_over_column`.

Reply:
Yes, "binary functions, string functions, window functions" all are named consistently with "column functions". But this is not only a matter of consistency, there's actually an important difference between "with column" and "over column". They mean different things.

Reply:
okie. Let me fix it to over.